src/mlpack/methods/ann/rbm.hpp

```diff
@@ -0,0 +1,243 @@
+/**
+ * @file vanilla_rbm.hpp
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_VANILLA_RBM_HPP
+#define MLPACK_METHODS_ANN_VANILLA_RBM_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+#include "layer/layer.hpp"
+#include "layer/base_layer.hpp"
+
+#include "activation_functions/softplus_function.hpp"
+#include "init_rules/gaussian_init.hpp"
+#include "init_rules/random_init.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InitializationRuleType,
+    typename VisibleLayerType,
+    typename HiddenLayerType>
+class RBM
+{
+ public:
+  using NetworkType = RBM<InitializationRuleType, VisibleLayerType,
+      HiddenLayerType>;
+
+  /* 
+   * Intalise all the parameters of the network
+   * using the intialise rule. 
+   *
+   * @tparam: IntialiserType rule to intialise the parameters of the network
+   * @tparam: VisibleLayerType visible layer 
+   * @tparam: HiddenLyaerType hidden layer
+   */
+  RBM(arma::mat predictors, InitializationRuleType initializeRule,
```

Comment:
Looking at line 92 constructor of the FFN class
FFN(arma::mat predictors,
      arma::mat responses,
      OutputLayerType outputLayer = OutputLayerType(),
      InitializationRuleType initializeRule = InitializationRuleType());